habitat/sims/habitat_simulator/habitat_simulator.py

```diff
@@ -192,74 +203,105 @@ class HabitatSim(habitat_sim.Simulator, Simulator):
 
     def __init__(self, config: Config) -> None:
         self.habitat_config = config
-        agent_config = self._get_agent_config()
-
-        sim_sensors = []
-        for sensor_name in agent_config.SENSORS:
-            sensor_cfg = getattr(self.habitat_config, sensor_name)
-            sensor_type = registry.get_sensor(sensor_cfg.TYPE)
-
-            assert sensor_type is not None, "invalid sensor type {}".format(
-                sensor_cfg.TYPE
-            )
-            sim_sensors.append(sensor_type(sensor_cfg))
-
-        self._sensor_suite = SensorSuite(sim_sensors)
-        self.sim_config = self.create_sim_config(self._sensor_suite)
+        self.__sensor_suites: Dict[
+            int, SensorSuite
+        ] = self._regen_sensor_suites()
+        self.sim_config = self.create_sim_config(self.__sensor_suites)
         self._current_scene = self.sim_config.sim_cfg.scene_id
         super().__init__(self.sim_config)
+        # TODO Make action space MultiAgent compatible
         self._action_space = spaces.Discrete(
             len(self.sim_config.agents[0].action_space)
         )
         self._prev_sim_obs: Optional[Observations] = None
+        self._load_agent_body_meshes()
 
     def create_sim_config(
-        self, _sensor_suite: SensorSuite
+        self, _sensor_suite: Dict[int, SensorSuite]
     ) -> habitat_sim.Configuration:
         sim_config = habitat_sim.SimulatorConfiguration()
         # Check if Habitat-Sim is post Scene Config Update
         if not hasattr(sim_config, "scene_id"):
             raise RuntimeError(
                 "Incompatible version of Habitat-Sim detected, please upgrade habitat_sim"
             )
+
         overwrite_config(
             config_from=self.habitat_config.HABITAT_SIM_V0,
             config_to=sim_config,
         )
         sim_config.scene_id = self.habitat_config.SCENE
-        agent_config = habitat_sim.AgentConfiguration()
-        overwrite_config(
-            config_from=self._get_agent_config(), config_to=agent_config
-        )
-
-        sensor_specifications = []
-        for sensor in _sensor_suite.sensors.values():
-            sim_sensor_cfg = habitat_sim.SensorSpec()
+        agent_configs = []
+        for agent_id, sensor_suite in self.__sensor_suites.items():
+            agent_config = habitat_sim.AgentConfiguration()
             overwrite_config(
-                config_from=sensor.config, config_to=sim_sensor_cfg
+                config_from=self._get_agent_config(agent_id),
+                config_to=agent_config,
+                ignore_keys={"body_mesh_config"},
```

Comment:
Looks like `ignore_keys` hack mixes Simulator Agent config and Habitat Lab layer of agent specification. While `body_mesh_config` isn't supported as Agent's config part let's have it separately from HSIM config.

Reply:
I don't follow what you mean? The config is part of the AGENT_0 YAML subconfig.